### DISCUSSION TITLE
Get-only or Set-only Property

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/csharplang#7760] [Language Ideas]
Sometimes, we want to define properties that only contain either a `get` or `set` method, but we cannot achieve this with a single-line expression. Consequently, we have to write nested curly braces, as shown below.
```
float MyProperty
{
    get
    {
        float result = 0f;
        // ...
        return result;
    }
}
```
In this example, the `get` and its associated pair of curly braces together occupy three lines. Compared to methods, properties with such structure contain more redundancy, while the original purpose of properties is to be simple. But with a small modification, it can be simplified, as shown below.
```
float get MyProperty
{
    float result = 0f;
    // ...
    return result;
}
```
Or like this.
```
float MyProperty get
{
    float result = 0f;
    // ...
    return result;
}
```
Similarly, set-only properties are also like this. Since get-only properties are more common, the `get` can be further omitted, like this:
```
float MyProperty
{
    float result = 0f;
    // ...
    return result;
}
```

**Comment by yaakov-h** [comment]
If line count is all you are worried about then this can be done today.

```csharp
float MyProperty { get
{
    float result = 0f;
    // ...
    return result;
}}
```

**Reply by YT-GameMaker** [reply]
The focus is not on the number of lines but on reducing unnecessary redundancy to enhance readability.

**Reply by huoyaoyuan** [reply]
I don't think removing the `get` declaration improves readability.

**Reply by DavidArno** [reply]
@yaakov-h, or it could be taken to the extreme:
```cs
float MyProperty{get{var result=0f;...;return result;}}
```
The point being that ideas such as that proposed by the OP are about reducing line count without compromising readability, so "_just make you code unreadable_" comments like yours really don't really add anything useful to the discussion.

**Reply by YT-GameMaker** [reply]
@huoyaoyuan No `get` means `get`, if it is a rule, then there won't be any disagreement, just as `private` can be omitted, or like the `get` that can be omitted in a get expression.

**Comment by HaloFour** [comment]
With [sequence expressions](https://github.com/dotnet/csharplang/issues/377) you could use a normal expression-bodied property without requiring a new form of syntax specific to this purpose.

**Comment by CyrusNajmabadi** [comment]
Very unlikely to do this, given that a get-only method with statements is already extremely tiny and lightweight.

**Comment by DavidArno** [comment]
If expression blocks were ever implemented (#3086), then your getter example would be expressible as:
```cs
float MyProperty => {
    float result = 0f;
    // ...
    result
};
```

But from what I can remember, the language team seemed only interested in expression blocks for switch expressions and even there the interest seemed minimal, so I doubt this will happen.
